lib/portage/__init__.py

```diff
@@ -218,21 +218,23 @@
 }
 
 
-def _decode_argv(argv):
+def _decode_argv(argv: List[str]):
     # With Python 3, the surrogateescape encoding error handler makes it
     # possible to access the original argv bytes, which can be useful
     # if their actual encoding does no match the filesystem encoding.
     fs_encoding = sys.getfilesystemencoding()
     return [_unicode_decode(x.encode(fs_encoding, "surrogateescape")) for x in argv]
 
 
-def _unicode_encode(s, encoding=_encodings["content"], errors="backslashreplace"):
+def _unicode_encode(
```

Comment:
I dare say getting rid of these recoding functions would be the real final boss. They're relicts of py2 support.

Reply:
@kenrap I think as part of your work (not this PR / for now), we should consider looking at https://github.com/gentoo/portage/pull/700 and slaying that dragon.